lib/src/ast_node_pattern.c

```diff
@@ -86,6 +86,29 @@ const cypher_astnode_t *cypher_ast_node_pattern_get_identifier(
 }
 
 
+void cypher_ast_node_pattern_set_identifier(cypher_astnode_t *astnode,
+        cypher_astnode_t *identifier)
+{
+    REQUIRE_TYPE(astnode, CYPHER_AST_NODE_PATTERN, NULL);
+    REQUIRE_TYPE(identifier, CYPHER_AST_IDENTIFIER, NULL);
+    struct node_pattern *node =
+        container_of(astnode, struct node_pattern, _astnode);
+    assert(node->identifier == NULL && identifier != NULL);
+    node->identifier = identifier;
+    unsigned int n = astnode->nchildren;
+    if(n == 0)
+    {
+        astnode->children = malloc(sizeof(cypher_astnode_t *));
+    }
+    else
+    {
+        astnode->children =
+            realloc(astnode->children, (n + 1) * sizeof(cypher_astnode_t *));
```

Comment:
It's a bit pedantic, but I like to ensure there's always a check for a failed `malloc`/`realloc` (check for `NULL`) and then return an appropriate error indicator and set `errno`.

Reply:
No problem, I'll add that tomorrow!